### PR TITLE
docs(card): add promo docs to style and guidelines

### LINKED ISSUE #1818: [docs] Add info about promo variant of `<rh-card>`
### Description

Need to add some info about the promo variant of `<rh-card>` on the Style and Guideline subpages. Currently the only promo info we have on ux.redhat.com appears under demos.

### Links

[Figma file](https://www.figma.com/design/sWD636GlCTJOTQrJb82cV0/Element---Card---Promo?node-id=1-3&t=aitoWXdT2qbB2X8M-1)

### Screenshots

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by adamjohnson**
## What I did

1. Added Promo docs to Style page:
    * Variants
    * Theme
    * Responsive Design
    * Spacing
1. Added promo docs to Guidelines page:
    * Variants
    * Best Practices
1. Cleaned up the card grid on the Guidelines page

## Testing Instructions

1. Obtain a _card_ to write down feedback
2. Open the [Style page](https://deploy-preview-1956--red-hat-design-system.netlify.app/elements/card/style/#variants)
3. Make note of any differences from the Figma
4. Write any feedback on aforementioned card
5. Open the [Guidelines page](https://deploy-preview-1956--red-hat-design-system.netlify.app/elements/card/guidelines/#variants-1)
6. Make note of any differences to the Figma
7. Inscribe Guidelines notes on the backside of _card_.
8. Summon carrier pigeon.
9. Attach and send carrier pigeon to Carrier Pigeon Center™️  in southern Appalachia. ;-) 

[Figma](https://www.figma.com/design/sWD636GlCTJOTQrJb82cV0/Element---Card---Promo?node-id=1-3&t=aitoWXdT2qbB2X8M-1) to reference.

## Notes to Reviewers

  * Many images did not have the same height; thus, the Do/Don't heights are different and don't quite line up.
  * This PR combines Promo docs with existing docs. Make sure:
    * Heading levels are correct
    * Content is inserted where it should be
    * Content isn't duplicated (old and/or/vs new)

Closes #1818.

### CODE REVIEW

File: elements/rh-card/docs/10-style.md [hunk 1]
```diff
@@ -174,3 +182,90 @@ become thinner, the container padding decreases.
...
+  <img src="../card-variants-promo-standard.svg" alt="Standard promo with placeholder text">
+</uxdot-example>
+
+#### Featured promo
+
+A Featured promo uses the `Lightest` or `Darkest` surface color with a subtle border. An image can appear on the left or right side of a promo, but a featured promo can also omit an image.
+
+<uxdot-example>
+  <img src="../card-variants-promo-featured.svg" alt="Featured promo with placeholder text and image">
+</uxdot-example>
```
Comment (zeroedin) [inline elements/rh-card/docs/10-style.md:206]: Use the following with correct `[values]` just to prep for the merge of PR later that fixes this very thing.
```Suggestion
<uxdot-example width-adjustment="[value]px">
  <img src="../card-variants-promo-featured.svg" 
      alt="Featured promo with placeholder text and image"
      width="[value]"
      height="[value]">
</uxdot-example>
```

File: elements/rh-card/docs/10-style.md [hunk 2]
```diff
@@ -174,3 +182,90 @@ become thinner, the container padding decreases.
...
+
+#### Full-width promo
+
+A Full-width promo bleeds to the left and right edges of the screen. It does not have a border. Similarly to a featured promo, a full-width promo can use either the `Lightest` or `Darkest` surface color, and it can have no image or an image positioned to the left or right of the text.
+
+<uxdot-example variant="full">
+  <img src="../card-variants-promo-full-width.svg" alt="Full-width promo with placeholder text and image">
+</uxdot-example>
```
Comment (zeroedin) [inline elements/rh-card/docs/10-style.md:214]: same as above add width and height to img and width-adjustment to uxdot-example to match

File: elements/rh-card/docs/10-style.md [hunk 3]
```diff
@@ -174,3 +182,90 @@ become thinner, the container padding decreases.
...
+
+#### Narrow promo
+
+A Narrow promo can fit into a section or container smaller than 296 pixels. It moves an image in any type of promo below the body text and call to action. This is also the configuration used on phone viewports for a featured, standard promo, or full-width promo.
+
+<uxdot-example width-adjustment="752px">
+  <img src="../card-variants-promo-narrow.svg" alt="Narrow promo versions with featured promo and standard promo">
+</uxdot-example>
```
Comment (zeroedin) [inline elements/rh-card/docs/10-style.md:222]: again

File: elements/rh-card/docs/10-style.md [hunk 4]
```diff
@@ -174,3 +182,90 @@ become thinner, the container padding decreases.
...
+
+### Theme
+
+#### Promo
+
+The standard promo uses the `Lighter` (in light theme) and `Darker` (in dark theme) surface colors.
+
+<uxdot-example>
+  <img src="../card-theme-promo-standard.svg" alt="Standard Promo's in light and dark themes">
+</uxdot-example>
```
Comment (zeroedin) [inline elements/rh-card/docs/10-style.md:232]: etc.

File: elements/rh-card/docs/10-style.md [hunk 5]
```diff
@@ -174,3 +182,90 @@ become thinner, the container padding decreases.
...
+
+The featured and full-width promos use the `Lightest` and `Darkest` surface colors for light and dark theme, respectively.
+
+<uxdot-example no-border variant="full">
+  <img src="../card-theme-promo-featured-full-width.svg" alt="Featured and full-width Promo's in light and dark themes">
+</uxdot-example>
```
Comment (zeroedin) [inline elements/rh-card/docs/10-style.md:238]: etc

File: elements/rh-card/docs/10-style.md [hunk 6]
```diff
@@ -174,3 +182,90 @@ become thinner, the container padding decreases.
...
+
+### Spacing
+
+#### Promo
+
+Except for narrow promos, spacing within the other promo configurations is a little different from Card.
+
+<uxdot-example class="promo-spacing" alignment="left">
+  <figure>
+    <img src="../card-spacing-promo-standard.svg" alt="Standard promo with spacing annotations. 48px on all sides and 24px to the left of the CTA.">
```
Comment (zeroedin) [inline elements/rh-card/docs/10-style.md:254]: add width and height values

File: elements/rh-card/docs/10-style.md [hunk 7]
```diff
@@ -174,3 +182,90 @@ become thinner, the container padding decreases.
...
+    <figcaption>Standard promo</figcaption>
+  </figure>
+  <figure>  
+    <img src="../card-spacing-promo-featured.svg" alt="Featured promo with spacing annotations. 48px on all sides. 32px between the copy and image. 16px after the heading and 24px above the CTA.">
+    <figcaption>Featured promo</figcaption>
+  </figure>
+  <figure>
+    <img src="../card-spacing-promo-full-width.svg" alt="Full-width promo with spacing annotations. 32px between the image and body copy.">
```
Comment (zeroedin) [inline elements/rh-card/docs/10-style.md:262]: add width and height values

File: elements/rh-card/docs/10-style.md [hunk 8]
```diff
@@ -174,3 +182,90 @@ become thinner, the container padding decreases.
...
+    <figcaption>Full-width promo</figcaption>
+  </figure>
+  <figure>
+    <div style="max-width: 360px;">
+      <img src="../card-spacing-promo-narrow.svg" alt="Narrow promo with spacing annotations. 32px on all sides and 24px above the CTA.">
```
Comment (zeroedin) [inline elements/rh-card/docs/10-style.md:267]: add width and height values

File: elements/rh-card/docs/20-guidelines.md [hunk 9]
```diff
@@ -236,28 +229,99 @@ place any other complex components in card layouts.
...
+## Best practices
+
+### Card width
+
+<uxdot-best-practice variant="do">
+  <uxdot-example slot="image">
+    <img src="../card-best-practices-width-do.svg" alt="Three cards in a row">
+  </uxdot-example>
+  <p>In a 12-column layout, use a minimum width of four columns for a card. The maximum number of cards that should be used in a row is three.</p>
+</uxdot-best-practice>
```
Comment (zeroedin) [inline elements/rh-card/docs/20-guidelines.md:263]: rinse and repeat.  you know the drill by now :)

**Review by bennypowers — APPROVED** [review]
I made some edits to the text for brevity, but more importantly, I tried to disambiguate the terms "theme" and "color-palette" - here, as in the theming pages, we should use the term "color palette" to refer to the color-palette attribute, and "background" to refer to the color context. 

Note that there is no mention in this document of "theming" - meaning the process of customizing the element through css props and parts

In the interest of avoiding confusion and encouraging adoption and contribution, we should rally around these three definitions:
- **Background** - represents the `on` colors, when a consumer receives context from it's provider, can be one of `light` or `dark`, defaults to `light`
- **Color palette** - refers to the six color palettes - `(light|dark)(er|est)?`, or a subset thereof. Represents the `color-palette` attribute, which if set on a provider, determines the context for it's children.
- **Theming** - the developer process of customizing the color system by way of css custom properties (and to a lesser extent css shadow parts)

**Comment by marionnegp** [comment]
> By this I mean that we should no longer - anywhere in our docs or examples - refer to "light theme" or "dark theme", but rather "light background" or "dark background", and we should also explicitly distinguish between light/dark backgrounds and (light|dark)(er|est)? color palettes. When we _do_ use the word "theme", it should only be in reference to the developer process of customizing the design system's color subsystem via css props.

I like being able to distinguish between the definitions, but I think it's going to be difficult to have non-developers adjust how they use "theme" and "theming" and consider theming as only a developer process.

**Comment by bennypowers** [comment]
> ... non-developers ... only a developer process.

Good point. Is this something we can fix with more and better words? Ultimately we're producing HTML, and the closer design and dev are regarding terminology and shared knowledge of how the DS works, the better.

To wit: Designers also need to know about overriding theme tokens in order to create a theme, so in that regard, even though we don't yet have the facility to compute contexts from color-palettes in figma, the idea of theming as a process can also apply to designers.

**Comment by bennypowers** [comment]
Thinking it over again, the important part for me isn't that theming is for developers, it's rather that:

* Color palette is a choice, it's something you set actively
" Background (fka context, fka on) is an effect, it's a consequence of having a color palette (or having a default color palette)
* Theming is a process: it's the act of overriding theme tokens with custom values 

I think both designers and developers need to know those things

**Comment by bennypowers** [comment]
demoted to draft in light of above discussion

**Review by marionnegp — APPROVED** [review]
The edits @bennypowers made to the color/theming terminology sounds good! Thanks for bearing with my questions. 😅

File: elements/rh-card/docs/10-style.md [hunk 10]
```diff
@@ -174,3 +182,120 @@ become thinner, the container padding decreases.
 <uxdot-example width-adjustment="360px">
   <img src="../card-spacing-mobile.svg" alt="Card spacing on mobile">
 </uxdot-example>
+
+## Variants
```
[the remark this answers is not in the document]
Comment (adamjohnson) [reply elements/rh-card/docs/10-style.md]: Looks like you took care of this, nice job!

File: elements/rh-card/docs/20-guidelines.md [hunk 11]
```diff
@@ -45,7 +38,7 @@ Default calls to action may be used.
```
[the remark this answers is not in the document]
Comment (adamjohnson) [reply elements/rh-card/docs/20-guidelines.md:37]: Looks like you took care of this as well. 👏

**Pull request merged**